packages/typst.node/src/compiler/boxed.rs

```diff
@@ -108,35 +112,28 @@ impl BoxedCompiler {
             }
         };
 
-        let Some(new_state) = new_state else {
-            return Ok(None);
-        };
-
-        world
-            .mutate_entry(new_state)
-            .map(Some)
-            .map_err(map_node_error)
+        Ok(self.universe.snapshot_with(Some(TaskInputs {
+            entry: new_state,
+            ..Default::default()
+        })))
     }
 
     pub fn compile_raw(
         &mut self,
         compile_by: CompileDocArgs,
     ) -> napi::Result<NodeTypstCompileResult, NodeError> {
-        let e = self.setup_compiler_by(compile_by)?;
+        let world = self.setup_compiler_by(compile_by)?;
 
         let res = if self.0.universe().entry_state().is_inactive() {
             Err(map_node_error(error_once!("entry file is not set")))
```

Comment:
we can directly return errors here, since no state is to recover:
```patch
- if let Some(entry_file) = e {
-     self.0
-         .universe_mut()
-         .mutate_entry(entry_file)
-         .map_err(map_node_error)?;
- }
```

Reply:
I did it by myself in [f8da011](https://github.com/Myriad-Dreamin/typst.ts/pull/550/commits/f8da011b99bdfc606a140cd8b93c383e2887990f)

Reply:
I see > <